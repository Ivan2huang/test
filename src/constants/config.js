import getConfig from 'next/config';

const { publicRuntimeConfig } = getConfig();

const CONFIG = {
  ...publicRuntimeConfig,
  dateFormat: 'd MMM yyyy',
  dateCustomChineseFormat: "yyyy'年'M'月'dd'日'",
  monthDayCustomChineseFormat: "M'月'dd'日'",
  monthDayCustomEnglishFormat: 'd MMM',
  dateCustomThaiFormat: 'ddวัน mmเดือน yyyyปี',
  maxRecentSearch: 4,
  locales: {
    ENGLISH: 'en-HK',
    CHINESE: 'zh-HK',
    THAI: 'th-TH',
  },
  fakeData: false,
};

export const enabledFeatureToggledForClusteredMap = () =>
  publicRuntimeConfig.featureToggleClusteredMap;

export default CONFIG;
