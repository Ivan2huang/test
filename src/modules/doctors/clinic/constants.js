const CLINIC_LIST_VIEW_TYPE = {
  selectedStackedClinics: 'selectedStackedClinics',
  filteredClinics: 'filteredClinics',
};

const FILTER_BY = {
  consultationType: 'consultationType',
};

const VIEW = {
  list: 'list',
  map: 'map',
};

export { CLINIC_LIST_VIEW_TYPE, FILTER_BY, VIEW };
