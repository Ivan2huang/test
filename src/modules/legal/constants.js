// eslint-disable-next-line import/prefer-default-export
export const LOCALE_STORAGE = 'LOCALE_STORAGE';
