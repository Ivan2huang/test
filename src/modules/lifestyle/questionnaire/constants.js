export const QUESTIONNAIRE_TYPES = {
  aboutMe: 'aboutMe',
  choices: 'choices',
  health: 'health',
  futureMe: 'futureMe',
};

export const BMI_UNIT = {
  weight: {
    KG: 'Kg',
    LB: 'Lb',
  },
  height: {
    CM: 'Cm',
    IN: 'In',
    FTIN: 'FtIn',
  },
};

export const UPLOAD_RULE_ID = 'upload-rule';

export const QUESTIONNAIRE_FORM = 'lifestyle-questionnaire';
