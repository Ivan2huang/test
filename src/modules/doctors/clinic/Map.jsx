/* eslint-disable react/no-this-in-sfc */
import React, { useEffect, useState } from 'react';
import { injectIntl } from 'react-intl';
import PropTypes from 'prop-types';
import { MAP } from 'react-google-maps/lib/constants';

import {
  GoogleMap,
  Marker,
  withGoogleMap,
  withScriptjs,
} from 'react-google-maps';
import { Box, Hidden } from '@material-ui/core';

import Images from '../../../constants/images';
import CustomMapControl from './CustomMapControl';
import {
  DEFAULT_LOCATION,
  DEFAULT_MAP_OPTIONS,
  DEFAULT_ZOOM,
  MAP_CONTROL_POSITIONS,
  MINIMUM_ZOOM_ON_SEARCH,
} from './mapConfig';
import Toast from '../../../uiComponents/Toast';
import { formatMessage } from '../../../helpers/helpers';
import checkNetworkConnectivity from '../../../helpers/networkCheck';
import ClusterMapMarker from './ClusterMapMarker';
import { enabledFeatureToggledForClusteredMap } from '../../../constants/config';
import { VIEW } from './constants';
import NonClusterMapMarkerContainer from './NonClusterMapMarkerContainer';

const MAX_ZOOM_FOR_CLUSTERING = 22;

const Map = (
  {
    intl,
    clinics,
    view,
    showClinicDetail,
    selectedClinic,
    fitMapToOriginalPosition,
    selectedClinicFromSearchResultPanel,
    setFitMapToOriginalPosition,
  },
  context,
) => {
  const [currentLocation, setCurrentLocation] = useState(DEFAULT_LOCATION);
  const [showMyCurrentLocation, setShowMyCurrentLocation] = useState(false);
  const [openSnackbar, setOpenSnackbar] = useState(false);
  const [showClusterMarker, setShowClusterMarker] = useState(true);
  // eslint-disable-next-line react/destructuring-assignment
  const map = context[MAP];
  const closeSnackBar = () => {
    setOpenSnackbar(false);
  };

  const openSnackBar = () => {
    setOpenSnackbar(true);
  };

  const success = position => {
    setShowMyCurrentLocation(true);
    setCurrentLocation({
      lat: position.coords.latitude,
      lng: position.coords.longitude,
    });
  };

  const error = () => {
    setShowMyCurrentLocation(false);
    openSnackBar();
  };

  const handleZoomChanged = () => {
    const currentZoom = map.getZoom();
    setShowClusterMarker(currentZoom < MAX_ZOOM_FOR_CLUSTERING);
  };

  const centerToMyLocation = () => {
    const location = window.navigator && window.navigator.geolocation;
    if (location) {
      location.getCurrentPosition(success, error);
    } else {
      openSnackBar();
    }
  };

  const fitAllClinicsToMapScreen = () => {
    if (showMyCurrentLocation) setShowMyCurrentLocation(false);
    setTimeout(() => {
      if (clinics && clinics.length) {
        const { google } = window;
        const bounds = new google.maps.LatLngBounds();
        clinics.forEach(clinic => {
          if (clinic.latitude && clinic.longitude) {
            bounds.extend({ lat: clinic.latitude, lng: clinic.longitude });
          }
        });
        map.fitBounds(bounds);
        if (map.getZoom() < 2) {
          const center = map.getCenter();
          map.setZoom(MINIMUM_ZOOM_ON_SEARCH);
          map.setCenter(center);
        }
      }
    });
  };

  if (fitMapToOriginalPosition) {
    fitAllClinicsToMapScreen();
    setFitMapToOriginalPosition(false);
  }

  useEffect(() => {
    checkNetworkConnectivity();
  }, []);

  useEffect(() => {
    fitAllClinicsToMapScreen();
  }, [clinics]);

  useEffect(() => {
    if (!showClinicDetail) {
      fitAllClinicsToMapScreen();
      setCurrentLocation(DEFAULT_LOCATION);
    }
  }, [showClinicDetail]);

  useEffect(() => {
    if (showMyCurrentLocation) setShowMyCurrentLocation(false);
    if (!enabledFeatureToggledForClusteredMap()) {
      if (selectedClinic.latitude && selectedClinic.longitude) {
        setCurrentLocation({
          lat: selectedClinic.latitude,
          lng: selectedClinic.longitude,
        });
      }
    }
    if (
      enabledFeatureToggledForClusteredMap() &&
      selectedClinicFromSearchResultPanel
    ) {
      if (selectedClinic.latitude && selectedClinic.longitude) {
        setCurrentLocation({
          lat: selectedClinic.latitude,
          lng: selectedClinic.longitude,
        });
        map.setZoom(MAX_ZOOM_FOR_CLUSTERING);
      }
    }
  }, [selectedClinic]);

  return (
    <Box
      implementation="css"
      smDown={showClinicDetail || view === VIEW.list}
      component={Hidden}
    >
      <GoogleMap
        defaultCenter={DEFAULT_LOCATION}
        defaultZoom={DEFAULT_ZOOM}
        defaultOptions={DEFAULT_MAP_OPTIONS}
        center={currentLocation}
        onZoomChanged={() => {
          handleZoomChanged();
        }}
      >
        {showMyCurrentLocation && (
          <Marker
            index="current-location"
            position={currentLocation}
            options={{ icon: { url: `${Images.LOCATION}` } }}
          />
        )}
        {enabledFeatureToggledForClusteredMap() ? (
          <ClusterMapMarker
            clinics={clinics}
            showClusterMarker={showClusterMarker}
          />
        ) : (
          <NonClusterMapMarkerContainer clinics={clinics} />
        )}

        <CustomMapControl position={MAP_CONTROL_POSITIONS.myLocationIcon}>
          {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-noninteractive-element-interactions */}
          <img
            data-testid="img-my-location"
            src={Images.MY_LOCATION}
            alt="My Location"
            onClick={centerToMyLocation}
          />
        </CustomMapControl>
      </GoogleMap>
      <Toast
        open={openSnackbar}
        handleClose={closeSnackBar}
        message={formatMessage(
          intl,
          'lifestyle.clinic.snackbar.message',
          'Google Maps does not have permission to use your location. Enable location services on your browser.',
        )}
      />
    </Box>
  );
};

Map.contextTypes = {
  [MAP]: PropTypes.shape({}),
};

Map.defaultProps = {
  selectedClinic: {},
};

Map.propTypes = {
  intl: PropTypes.shape({}).isRequired,
  clinics: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  view: PropTypes.string.isRequired,
  showClinicDetail: PropTypes.bool.isRequired,
  selectedClinic: PropTypes.shape({}),
  fitMapToOriginalPosition: PropTypes.bool.isRequired,
  selectedClinicFromSearchResultPanel: PropTypes.bool.isRequired,
  setFitMapToOriginalPosition: PropTypes.func.isRequired,
};

export default injectIntl(withScriptjs(withGoogleMap(Map)));
