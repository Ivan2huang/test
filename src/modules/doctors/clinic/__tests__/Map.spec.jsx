/* eslint-disable react/prop-types, jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions,react/self-closing-comp,react/button-has-type,global-require,no-return-assign,react/jsx-one-expression-per-line */
import React from 'react';
import { MAP } from 'react-google-maps/lib/constants';
import { fireEvent, render, wait } from '@testing-library/react';
import PropTypes from 'prop-types';

import Map from '../Map';
import withIntl from '../../../../i18n/withIntlProvider';
import { enabledFeatureToggledForClusteredMap } from '../../../../constants/config';

jest.mock('react-google-maps', () => ({
  withScriptjs: Component => props => (
    <div>
      With Script JS
      <Component {...props} />
    </div>
  ),
  withGoogleMap: Component => ({
    googleMapURL,
    loadingElement,
    containerElement,
    mapElement,
    ...rest
  }) => (
    <div>
      With Google Map
      {googleMapURL}
      {loadingElement}
      {containerElement}
      {mapElement}
      <Component {...rest} />
    </div>
  ),
  GoogleMap: ({ children, center, ...rest }) => (
    <div>
      {Object.entries(rest).map(([key, value]) =>
        key === 'onZoomChanged'
          ? `${key} = ${value.name}`
          : `${key} = ${value}`,
      )}
      Google Map
      {`Center (lat: ${center.lat}, lng: ${center.lng})`}
      {children}
      <div data-testid="dummyTestButton" onClick={rest.onZoomChanged} />
    </div>
  ),
  Marker: ({ index, position, options, onClick }) => (
    <div {...position} {...options}>
      Marker
      {options.icon.url}
      <div data-testid={`map-pin-${index}`} onClick={onClick} />
    </div>
  ),
}));

jest.mock('../CustomMapControl', () => ({ children, ...rest }) => (
  <div>
    Custom Map Control
    {Object.entries(rest).map(([key, value]) => `${key} = ${value}`)}
    {children}
  </div>
));

jest.mock(
  '../../../../uiComponents/Toast',
  () => ({ message, handleClose, open, ...rest }) => (
    <div data-testid="no-location-access-toast" data-open={open} {...rest}>
      {message}
      <div data-testid="no-location-access-toast-close" onClick={handleClose}>
        Close
      </div>
    </div>
  ),
);

jest.mock('../mapConfig', () => ({
  DEFAULT_LOCATION: {
    lat: 12,
    lng: 30,
  },
  DEFAULT_MAP_OPTIONS: {
    disableDefaultUI: true,
    zoomControl: true,
  },
  DEFAULT_ZOOM: 10,
  MINIMUM_ZOOM_ON_SEARCH: 2,
  MAP_CONTROL_POSITIONS: {
    myLocationIcon() {
      return 3;
    },
  },
}));

jest.mock('../../../../helpers/helpers', () => ({
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
}));

jest.mock('../../../../helpers/networkCheck', () => jest.fn(() => {}));

jest.mock('../ClusterMapMarker', () => props => (
  <div>
    Dummy ClusterMapComponent
    {Object.entries(props).map(([key, value]) => `${key} = ${value}`)}
  </div>
));

jest.mock('../NonClusterMapMarkerContainer', () => ({ clinics }) => (
  <div>Dummy NonClusterMapMarkerContainer with clinics :{clinics.length}</div>
));

jest.mock('../../../../constants/config', () => {
  return {
    __esModule: true,
    default: {
      defaultLanguage: 'en',
    },
    enabledFeatureToggledForClusteredMap: jest.fn(),
  };
});

const extend = jest.fn();
const fitBounds = jest.fn();
const getCenter = jest.fn();
const setCenter = jest.fn();
const getZoom = jest.fn(() => 5);
const setZoom = jest.fn();
class MapComponent extends React.Component {
  getChildContext() {
    return {
      [MAP]: {
        controls: [[]],
        fitBounds,
        getCenter,
        setCenter,
        getZoom,
        setZoom,
      },
    };
  }

  render() {
    const { children } = this.props;
    return <div>{children}</div>;
  }
}
MapComponent.childContextTypes = {
  [MAP]: PropTypes.shape({}),
};

describe('Map Component', () => {
  const props = {
    googleMapURL: `testing-map-url`,
    loadingElement: <div>Loading</div>,
    containerElement: <div>Container</div>,
    mapElement: <div>Map</div>,
    clinics: [
      {
        id: '123',
        name: 'Dr Test Clinic',
        latitude: 1.3006643,
        longitude: 103.8001065,
      },
      {
        id: '124',
        name: 'Dr Clinic Wang',
        latitude: 3.3006643,
        longitude: 109.8001065,
      },
    ],
    view: 'map',
    showClinicDetail: false,
    selectedClinic: {},
    fitMapToOriginalPosition: false,
    selectedClinicFromSearchResultPanel: false,
    setFitMapToOriginalPosition: jest.fn(),
  };
  afterEach(() => {
    jest.clearAllMocks();
  });

  let Component;
  const setup = (componentProps = props) => {
    const ComponentWithIntl = withIntl(Map);
    Component = p => (
      <MapComponent>
        <ComponentWithIntl {...p} />
      </MapComponent>
    );
    return render(<Component {...componentProps} />);
  };
  describe('with clustered map view feature toggale on', () => {
    beforeEach(() => {
      enabledFeatureToggledForClusteredMap.mockReturnValue(true);
    });
    it('should not fit bound clinic locations when there are no clinics', async () => {
      const newProps = {
        ...props,
        clinics: [],
      };
      setup(newProps);

      await wait(() => expect(fitBounds).toHaveBeenCalledTimes(0));
    });

    it('should fit map to original position when fitMapToOriginalPosition is true', async () => {
      const newProps = {
        ...props,
        fitMapToOriginalPosition: true,
      };
      window.google = {
        maps: {
          LatLngBounds: jest.fn().mockImplementation(() => {
            return { extend };
          }),
        },
      };
      setup(newProps);

      await wait(() => expect(getZoom).toHaveBeenCalledTimes(3));
      await wait(() =>
        expect(newProps.setFitMapToOriginalPosition).toHaveBeenCalled(),
      );
    });

    it('should fit map to original position when fitMapToOriginalPosition is true and clinic details enabled', async () => {
      const newProps = {
        ...props,
        showClinicDetail: true,
        fitMapToOriginalPosition: true,
      };
      window.google = {
        maps: {
          LatLngBounds: jest.fn().mockImplementation(() => {
            return { extend };
          }),
        },
      };
      setup(newProps);

      await wait(() => expect(getZoom).toHaveBeenCalledTimes(2));
      await wait(() =>
        expect(newProps.setFitMapToOriginalPosition).toHaveBeenCalled(),
      );
    });

    it('should call fit bound on map for given clinic locations', async () => {
      window.google = {
        maps: {
          LatLngBounds: jest.fn().mockImplementation(() => {
            return { extend };
          }),
        },
      };

      setup();

      await wait(() => expect(fitBounds).toHaveBeenCalledTimes(2));
    });

    it('should default to minimum zoom level on search if fitbound results in lesser zoom level', async () => {
      getZoom.mockReturnValue(1);
      window.google = {
        maps: {
          LatLngBounds: jest.fn().mockImplementation(() => {
            return { extend };
          }),
        },
      };

      setup();

      await wait(() => expect(getCenter).toHaveBeenCalledTimes(2));
      await wait(() => expect(setCenter).toHaveBeenCalledTimes(2));
      await wait(() => expect(getZoom).toHaveBeenCalledTimes(2));
      await wait(() => expect(setZoom).toHaveBeenCalledTimes(2));
      await wait(() => expect(setZoom).toHaveBeenCalledWith(2));
    });

    it('should match snapshot', () => {
      window.google = {
        maps: {
          LatLngBounds: jest.fn().mockImplementation(() => {
            return { extend };
          }),
        },
      };
      const { container, getByTestId } = setup();

      const ele = getByTestId('dummyTestButton');
      fireEvent.click(ele);

      expect(container).toMatchSnapshot();
    });

    it('should center to current Location on my location click', () => {
      window.navigator.geolocation = {
        getCurrentPosition: jest.fn(success =>
          success({ coords: { latitude: 12, longitude: 23 } }),
        ),
      };

      const { container, getByTestId } = setup();
      const myLocation = getByTestId('img-my-location');

      fireEvent.click(myLocation);

      expect(container).toMatchSnapshot();
    });

    it('should not show current location when clinics are updated', () => {
      const newProps = {
        ...props,
        clinics: [
          {
            id: '124',
            name: 'Dr Test Clinic',
            latitude: 1.3006643,
            longitude: 103.8001065,
          },
          {
            id: '125',
            name: 'Er Test Clinic',
            latitude: 2.3006643,
            longitude: 105.8001065,
          },
        ],
      };

      window.navigator.geolocation = {
        getCurrentPosition: jest.fn(success =>
          success({ coords: { latitude: 12, longitude: 23 } }),
        ),
      };

      const { container, getByTestId, rerender } = setup();
      const myLocation = getByTestId('img-my-location');

      fireEvent.click(myLocation);
      rerender(<Component {...newProps} />);

      expect(container).toMatchSnapshot();
    });

    it('should display the snack bar with message with location service denied', () => {
      window.navigator.geolocation = {
        getCurrentPosition: jest.fn((success, error) => error()),
      };
      const { container, getByTestId } = setup();
      const myLocation = getByTestId('img-my-location');

      fireEvent.click(myLocation);

      expect(container).toMatchSnapshot();
    });

    it('should close snackbar when it is closed by clickin on cross', () => {
      window.navigator.geolocation = {
        getCurrentPosition: jest.fn((success, error) => error()),
      };
      const { container, getByTestId } = setup();

      const centerToMyLocation = getByTestId('img-my-location');
      const closeToast = getByTestId('no-location-access-toast-close');
      const isToastOpen = getByTestId('no-location-access-toast').getAttribute(
        'data-open',
      );

      fireEvent.click(centerToMyLocation);
      fireEvent.click(closeToast);

      expect(isToastOpen).toBe('false');
      expect(container).toMatchSnapshot();
    });

    it('should not show current location when clinic is selected', () => {
      window.navigator.geolocation = {
        getCurrentPosition: jest.fn(success =>
          success({ coords: { latitude: 12, longitude: 23 } }),
        ),
      };
      const newProps = {
        ...props,
        selectedClinic: {
          id: '123',
          name: 'Dr Test Clinic',
          latitude: 1.3006643,
          longitude: 103.8001065,
        },
      };
      const { container, rerender, getByTestId } = setup();

      // show current location
      const myLocation = getByTestId('img-my-location');
      fireEvent.click(myLocation);

      // un-select selected clinic
      rerender(<Component {...newProps} />);

      // snapshot shouldn't have show current location marker
      expect(container).toMatchSnapshot();
    });

    it('should not center to current location when location is undefined', () => {
      Object.defineProperty(global, 'navigator', {
        value: undefined,
      });
      const { container, getByTestId } = setup();
      const myLocation = getByTestId('img-my-location');

      fireEvent.click(myLocation);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when a clinic is selected', () => {
      window.google = {
        maps: {
          LatLngBounds: jest.fn().mockImplementation(() => {
            return { extend };
          }),
        },
      };

      const newProps = {
        ...props,
        selectedClinic: {
          id: '123',
          name: 'Dr Test Clinic',
          latitude: 1.3006643,
          longitude: 103.8001065,
        },
        selectedClinicFromSearchResultPanel: true,
      };
      const { container } = setup(newProps);

      expect(container).toMatchSnapshot();
      expect(setZoom).toHaveBeenCalledWith(22);
    });

    it('should match snapshot when a clinic is selected and selected clinic is not present in all clinics', () => {
      window.google = {
        maps: {
          LatLngBounds: jest.fn().mockImplementation(() => {
            return { extend };
          }),
        },
      };

      const newProps = {
        ...props,
        selectedClinic: {
          id: '126',
          name: 'Dr Not present',
          latitude: 6.3006643,
          longitude: 113.8001065,
        },
      };
      const { container } = setup(newProps);

      expect(container).toMatchSnapshot();
    });
  });
  describe('with clustered map view feature toggale off', () => {
    beforeEach(() => {
      enabledFeatureToggledForClusteredMap.mockReturnValue(false);
    });

    it('should match snapshot', () => {
      const { container } = setup();

      expect(container).toMatchSnapshot();
    });
  });
});
