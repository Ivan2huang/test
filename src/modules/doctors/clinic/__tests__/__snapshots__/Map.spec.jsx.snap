// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Component with clustered map view feature toggale off should match snapshot 1`] = `
<div>
  <div>
    <div>
      With Script JS
      <div>
        With Google Map
        testing-map-url
        <div>
          Loading
        </div>
        <div>
          Container
        </div>
        <div>
          Map
        </div>
        <div
          class="MuiBox-root MuiBox-root-240"
        >
          <div>
            defaultCenter = [object Object]
            defaultZoom = 10
            defaultOptions = [object Object]
            onZoomChanged = onZoomChanged
            Google Map
            Center (lat: 12, lng: 30)
            <div>
              Dummy NonClusterMapMarkerContainer with clinics :
              2
            </div>
            <div>
              Custom Map Control
              position = function myLocationIcon() {
        return 3;
      }
              <img
                alt="My Location"
                data-testid="img-my-location"
                src="/static/my_location.svg"
              />
            </div>
            <div
              data-testid="dummyTestButton"
            />
          </div>
          <div
            data-open="false"
            data-testid="no-location-access-toast"
          >
            Google Maps does not have permission to use your location. Enable location services on your browser.
            <div
              data-testid="no-location-access-toast-close"
            >
              Close
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Map Component with clustered map view feature toggale on should center to current Location on my location click 1`] = `
<div>
  <div>
    <div>
      With Script JS
      <div>
        With Google Map
        testing-map-url
        <div>
          Loading
        </div>
        <div>
          Container
        </div>
        <div>
          Map
        </div>
        <div
          class="MuiBox-root MuiBox-root-104"
        >
          <div>
            defaultCenter = [object Object]
            defaultZoom = 10
            defaultOptions = [object Object]
            onZoomChanged = onZoomChanged
            Google Map
            Center (lat: 12, lng: 23)
            <div
              icon="[object Object]"
              lat="12"
              lng="23"
            >
              Marker
              /static/location.png
              <div
                data-testid="map-pin-current-location"
              />
            </div>
            <div>
              Dummy ClusterMapComponent
              clinics = [object Object],[object Object]
              showClusterMarker = true
            </div>
            <div>
              Custom Map Control
              position = function myLocationIcon() {
        return 3;
      }
              <img
                alt="My Location"
                data-testid="img-my-location"
                src="/static/my_location.svg"
              />
            </div>
            <div
              data-testid="dummyTestButton"
            />
          </div>
          <div
            data-open="false"
            data-testid="no-location-access-toast"
          >
            Google Maps does not have permission to use your location. Enable location services on your browser.
            <div
              data-testid="no-location-access-toast-close"
            >
              Close
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Map Component with clustered map view feature toggale on should close snackbar when it is closed by clickin on cross 1`] = `
<div>
  <div>
    <div>
      With Script JS
      <div>
        With Google Map
        testing-map-url
        <div>
          Loading
        </div>
        <div>
          Container
        </div>
        <div>
          Map
        </div>
        <div
          class="MuiBox-root MuiBox-root-155"
        >
          <div>
            defaultCenter = [object Object]
            defaultZoom = 10
            defaultOptions = [object Object]
            onZoomChanged = onZoomChanged
            Google Map
            Center (lat: 12, lng: 30)
            <div>
              Dummy ClusterMapComponent
              clinics = [object Object],[object Object]
              showClusterMarker = true
            </div>
            <div>
              Custom Map Control
              position = function myLocationIcon() {
        return 3;
      }
              <img
                alt="My Location"
                data-testid="img-my-location"
                src="/static/my_location.svg"
              />
            </div>
            <div
              data-testid="dummyTestButton"
            />
          </div>
          <div
            data-open="false"
            data-testid="no-location-access-toast"
          >
            Google Maps does not have permission to use your location. Enable location services on your browser.
            <div
              data-testid="no-location-access-toast-close"
            >
              Close
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Map Component with clustered map view feature toggale on should display the snack bar with message with location service denied 1`] = `
<div>
  <div>
    <div>
      With Script JS
      <div>
        With Google Map
        testing-map-url
        <div>
          Loading
        </div>
        <div>
          Container
        </div>
        <div>
          Map
        </div>
        <div
          class="MuiBox-root MuiBox-root-138"
        >
          <div>
            defaultCenter = [object Object]
            defaultZoom = 10
            defaultOptions = [object Object]
            onZoomChanged = onZoomChanged
            Google Map
            Center (lat: 12, lng: 30)
            <div>
              Dummy ClusterMapComponent
              clinics = [object Object],[object Object]
              showClusterMarker = true
            </div>
            <div>
              Custom Map Control
              position = function myLocationIcon() {
        return 3;
      }
              <img
                alt="My Location"
                data-testid="img-my-location"
                src="/static/my_location.svg"
              />
            </div>
            <div
              data-testid="dummyTestButton"
            />
          </div>
          <div
            data-open="true"
            data-testid="no-location-access-toast"
          >
            Google Maps does not have permission to use your location. Enable location services on your browser.
            <div
              data-testid="no-location-access-toast-close"
            >
              Close
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Map Component with clustered map view feature toggale on should match snapshot 1`] = `
<div>
  <div>
    <div>
      With Script JS
      <div>
        With Google Map
        testing-map-url
        <div>
          Loading
        </div>
        <div>
          Container
        </div>
        <div>
          Map
        </div>
        <div
          class="MuiBox-root MuiBox-root-87"
        >
          <div>
            defaultCenter = [object Object]
            defaultZoom = 10
            defaultOptions = [object Object]
            onZoomChanged = onZoomChanged
            Google Map
            Center (lat: 12, lng: 30)
            <div>
              Dummy ClusterMapComponent
              clinics = [object Object],[object Object]
              showClusterMarker = true
            </div>
            <div>
              Custom Map Control
              position = function myLocationIcon() {
        return 3;
      }
              <img
                alt="My Location"
                data-testid="img-my-location"
                src="/static/my_location.svg"
              />
            </div>
            <div
              data-testid="dummyTestButton"
            />
          </div>
          <div
            data-open="false"
            data-testid="no-location-access-toast"
          >
            Google Maps does not have permission to use your location. Enable location services on your browser.
            <div
              data-testid="no-location-access-toast-close"
            >
              Close
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Map Component with clustered map view feature toggale on should match snapshot when a clinic is selected 1`] = `
<div>
  <div>
    <div>
      With Script JS
      <div>
        With Google Map
        testing-map-url
        <div>
          Loading
        </div>
        <div>
          Container
        </div>
        <div>
          Map
        </div>
        <div
          class="MuiBox-root MuiBox-root-206"
        >
          <div>
            defaultCenter = [object Object]
            defaultZoom = 10
            defaultOptions = [object Object]
            onZoomChanged = onZoomChanged
            Google Map
            Center (lat: 1.3006643, lng: 103.8001065)
            <div>
              Dummy ClusterMapComponent
              clinics = [object Object],[object Object]
              showClusterMarker = true
            </div>
            <div>
              Custom Map Control
              position = function myLocationIcon() {
        return 3;
      }
              <img
                alt="My Location"
                data-testid="img-my-location"
                src="/static/my_location.svg"
              />
            </div>
            <div
              data-testid="dummyTestButton"
            />
          </div>
          <div
            data-open="false"
            data-testid="no-location-access-toast"
          >
            Google Maps does not have permission to use your location. Enable location services on your browser.
            <div
              data-testid="no-location-access-toast-close"
            >
              Close
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Map Component with clustered map view feature toggale on should match snapshot when a clinic is selected and selected clinic is not present in all clinics 1`] = `
<div>
  <div>
    <div>
      With Script JS
      <div>
        With Google Map
        testing-map-url
        <div>
          Loading
        </div>
        <div>
          Container
        </div>
        <div>
          Map
        </div>
        <div
          class="MuiBox-root MuiBox-root-223"
        >
          <div>
            defaultCenter = [object Object]
            defaultZoom = 10
            defaultOptions = [object Object]
            onZoomChanged = onZoomChanged
            Google Map
            Center (lat: 12, lng: 30)
            <div>
              Dummy ClusterMapComponent
              clinics = [object Object],[object Object]
              showClusterMarker = true
            </div>
            <div>
              Custom Map Control
              position = function myLocationIcon() {
        return 3;
      }
              <img
                alt="My Location"
                data-testid="img-my-location"
                src="/static/my_location.svg"
              />
            </div>
            <div
              data-testid="dummyTestButton"
            />
          </div>
          <div
            data-open="false"
            data-testid="no-location-access-toast"
          >
            Google Maps does not have permission to use your location. Enable location services on your browser.
            <div
              data-testid="no-location-access-toast-close"
            >
              Close
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Map Component with clustered map view feature toggale on should not center to current location when location is undefined 1`] = `
<div>
  <div>
    <div>
      With Script JS
      <div>
        With Google Map
        testing-map-url
        <div>
          Loading
        </div>
        <div>
          Container
        </div>
        <div>
          Map
        </div>
        <div
          class="MuiBox-root MuiBox-root-189"
        >
          <div>
            defaultCenter = [object Object]
            defaultZoom = 10
            defaultOptions = [object Object]
            onZoomChanged = onZoomChanged
            Google Map
            Center (lat: 12, lng: 30)
            <div>
              Dummy ClusterMapComponent
              clinics = [object Object],[object Object]
              showClusterMarker = true
            </div>
            <div>
              Custom Map Control
              position = function myLocationIcon() {
        return 3;
      }
              <img
                alt="My Location"
                data-testid="img-my-location"
                src="/static/my_location.svg"
              />
            </div>
            <div
              data-testid="dummyTestButton"
            />
          </div>
          <div
            data-open="true"
            data-testid="no-location-access-toast"
          >
            Google Maps does not have permission to use your location. Enable location services on your browser.
            <div
              data-testid="no-location-access-toast-close"
            >
              Close
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Map Component with clustered map view feature toggale on should not show current location when clinic is selected 1`] = `
<div>
  <div>
    <div>
      With Script JS
      <div>
        With Google Map
        testing-map-url
        <div>
          Loading
        </div>
        <div>
          Container
        </div>
        <div>
          Map
        </div>
        <div
          class="MuiBox-root MuiBox-root-172"
        >
          <div>
            defaultCenter = [object Object]
            defaultZoom = 10
            defaultOptions = [object Object]
            onZoomChanged = onZoomChanged
            Google Map
            Center (lat: 12, lng: 23)
            <div>
              Dummy ClusterMapComponent
              clinics = [object Object],[object Object]
              showClusterMarker = true
            </div>
            <div>
              Custom Map Control
              position = function myLocationIcon() {
        return 3;
      }
              <img
                alt="My Location"
                data-testid="img-my-location"
                src="/static/my_location.svg"
              />
            </div>
            <div
              data-testid="dummyTestButton"
            />
          </div>
          <div
            data-open="false"
            data-testid="no-location-access-toast"
          >
            Google Maps does not have permission to use your location. Enable location services on your browser.
            <div
              data-testid="no-location-access-toast-close"
            >
              Close
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Map Component with clustered map view feature toggale on should not show current location when clinics are updated 1`] = `
<div>
  <div>
    <div>
      With Script JS
      <div>
        With Google Map
        testing-map-url
        <div>
          Loading
        </div>
        <div>
          Container
        </div>
        <div>
          Map
        </div>
        <div
          class="MuiBox-root MuiBox-root-121"
        >
          <div>
            defaultCenter = [object Object]
            defaultZoom = 10
            defaultOptions = [object Object]
            onZoomChanged = onZoomChanged
            Google Map
            Center (lat: 12, lng: 23)
            <div>
              Dummy ClusterMapComponent
              clinics = [object Object],[object Object]
              showClusterMarker = true
            </div>
            <div>
              Custom Map Control
              position = function myLocationIcon() {
        return 3;
      }
              <img
                alt="My Location"
                data-testid="img-my-location"
                src="/static/my_location.svg"
              />
            </div>
            <div
              data-testid="dummyTestButton"
            />
          </div>
          <div
            data-open="false"
            data-testid="no-location-access-toast"
          >
            Google Maps does not have permission to use your location. Enable location services on your browser.
            <div
              data-testid="no-location-access-toast-close"
            >
              Close
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
